File: client/src/features/orders/order.ts
export interface ShippingAddress {
  name: string;
  line1: string;
  line2?: string;
  city: string;
  state: string;
  postal_code: string;
  country: string;
}

export interface CreateOrderRequest {
  saveAddress: boolean;
  shippingAddress: ShippingAddress;
}

export interface OrderItem {
  productId: number;
  name: string;
  pictureUrl: string;
  price: number;
  quantity: number;
}

export interface PaymentSummary {
  last4: string;
  brand: string;
  expMonth: number;
  expYear: number;
}

export interface Order {
  id: number;
  buyerId: string;
  shippingAddress: ShippingAddress;
  orderDate: string;
  orderItems: OrderItem[];
  subtotal: number;
  deliveryFee: number;
  orderStatus: string;
  paymentSummary?: PaymentSummary;
  total: number;
}

File: API/Entities/OrderAggregate/Order.cs
namespace API.Entities.OrderAggregate;

public class Order
{
	public int Id { get; set; }
	public required string BuyerId { get; set; }
	public required ShippingAddress ShippingAddress { get; set; }
	public DateTime OrderDate { get; set; } = DateTime.UtcNow;
	public List<OrderItem> OrderItems { get; set; } = [];
	public long Subtotal { get; set; }
	public long DeliveryFee { get; set; }
	public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
	public string? PaymentIntentId { get; set; }
	public PaymentSummary? PaymentSummary { get; set; }

	public long GetTotal()
	{
		return Subtotal + DeliveryFee;
	}
}
